sympy/core/tests/test_arit.py

```diff
@@ -1519,7 +1519,14 @@ def test_Mod():
     assert (x + 3) % 1 == Mod(x, 1)
     assert (x + 3.0) % 1 == Mod(1.*x, 1)
     assert (x - S(33)/10) % 1 == Mod(x + S(7)/10, 1)
-    assert str(Mod(.6*x + y, .3*y)) == str(Mod(0.1*y + 0.6*x, 0.3*y))
```

Comment:
The reason why `str` cannot be used to compare (any) floating point numbers is that you need to use the `abs(x-y) < eps` idiom, since two floating point numbers `x` and `y` can be *equal* for the given accuracy `eps`, but their string representation can be different, e.g. 3.999 and 4 are equal for accuracy 1e-3. If you do `x==y` you are using machine precision for `eps`, which is too tight, since any arithmetic operation loses accuracy. The `str` comparison is using whatever accuracy `str` prints, which you have no easy control over (this is set at other places in the code), and it can still print 3.999 instead of 4 for 1e-3 accuracy, so it's not a good way either. 

Conclusion: the only way things can be compared is using the `abs(x-y) < eps`, which is exactly what I do below for this test.

To answer your second question, yes, that would probably work too.